### PR TITLE
Machine Learning - An overview

### PR DESCRIPTION
**Pull request by Saichethan**
My first article in section io

### CODE REVIEW

**Comment by hectorkambow** [comment]
Good afternoon @Saichethan thank you for your work on this article. Our technical review committee had a few more suggestions/edits for you before we push this to production:

Terms like "input-output pairs", "labels", "input", etc should be explained when introduced. Assume the reader is smart but has no prior exposure to the common terminology.

Line 45: "MNIST image classification" -- readers may not know what MNIST is. Say "Image classification where every image is associated with an accurate label of 'cat', 'dog', etc."

Line 46: stock prediction: Every blog on the internet says stock prediction, but there are unsupervised methods for stock prediction. What matters is: How is that supervised? What are the targets and labels?

Line 47-48: redundant references to clustering problems with no information. Add a sentence or 2 describing clustering instead of just writing "clustering"

Lines 47-49: As with clustering, the author is just writing words, not showing us anything.

Lines 68+ : We don't need the examples (line 69-74). At this point, the readers know the basics. They now need a complete example of ML workflow using the house price prediction example. Add a few sentences for each of the other workflow steps using that example so you show them what it means.

**Comment by Saichethan** [comment]
Updated

File: ML.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+### Flow
+
+1. We first understand user requirements
+2. Define an objective (Formulating problem)
+3. Collect data for training and evaluating
+4. learning algorthim generate a hypothesis function
+5. Hypothesis is used to predict on validation data if results are not satisfactory we try to enhance our hypothesis
+
+<p align="center">
```
Comment (hectorkambow) [inline ML.md:38]: @Saichethan  please refer to using .md (markdown) for
the syntax for that is :
 

Also going to include a brief YT video that really helped me with .md formatting. 
https://www.youtube.com/watch?v=HUBNt18RFbo

Also, make sure you are following new submission instructions as well - as all images will be placed within the same branch.
https://github.com/section-io/engineering-education/blob/master/UPLOAD_INSTRUCTIONS.md

Comment (hectorkambow) [reply ML.md:38]: HTML and other languages can be included as examples in code blocks if needed of course

**Comment by hectorkambow** [comment]
pr 101

**Pull request closed without merging**